src/engine/InputQuery.cpp

```diff
@@ -239,6 +238,31 @@ InputQuery &InputQuery::operator=( const InputQuery &other )
         }
     }
 
+    // Setting plConstraints and topological order
+    if ( !other._networkLevelReasoner )
+    {
+        for ( const auto &constraint : other._plConstraints )
+            _plConstraints.append( constraint->duplicateConstraint() );
+    }
+    else
+    {
+        for ( const auto &constraint : other._networkLevelReasoner->
+                  getConstraintsInTopologicalOrder() )
+        {
+            if ( !other._plConstraints.exists( constraint ) )
+                continue;
+
+            auto *newPlc = constraint->duplicateConstraint();
+            _plConstraints.append( newPlc );
+            _networkLevelReasoner->addConstraintInTopologicalOrder( newPlc );
+        }
+        for ( const auto &constraint : other._plConstraints )
```

Comment:
Same question - how can these two sets of constraints not be identical? (modulo the order of elements)

Reply:
This can happen if the following sequence happens:

inputQuery.constructNetworkLevelReasoner()
inputQuery.addPLConstraint( p )
newInputQuery = inputQuery

Reply:
Okay. So we just stick all the new ones at the end of the list? Does this happen in practice?

Reply:
As discussed offline, added an assertion to make sure the topological order contains all the plConstriants.